### PR TITLE
feat(sqs): add message group handler

### PR DESCRIPTION
**Pull request by MickVanDuijn**

### CODE REVIEW

**Comment by MickVanDuijn** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #156** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/156?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/skyleague/event-horizon/156?utm_source=stack-comment).

File: package.json [hunk 1]
```diff
@@ -54,17 +54,17 @@
...
     "@skyleague/event-horizon-dev": "^4.0.11",
-    "@skyleague/node-standards": "^3.10.5",
-    "@skyleague/therefore": "^3.0.2",
-    "@types/node": "^20.8.9",
+    "@skyleague/node-standards": "^4.0.0",
+    "@skyleague/therefore": "^3.1.0",
+    "@types/node": "^20.10.5",
     "aws-sdk-client-mock": "^3.0.0",
     "aws-sdk-client-mock-jest": "^3.0.0",
     "got": "^13.0.0",
```
Comment (pevisscher) [inline package.json:62]: v14 is OK now

File: src/events/sqs/sqs.spec.ts [hunk 2]
```diff
@@ -33,6 +34,23 @@ it('handles sqs events', async () => {
         expect(sqs).toHaveBeenCalledWith(expect.anything(), event.Records, ctx)
     })
 })
+it('handles sqs events - message grouping', async () => {
```
Comment (pevisscher) [inline src/events/sqs/sqs.spec.ts:37]:
```suggestion

it('handles sqs events - message grouping', async () => {
```

File: src/events/sqs/types.ts [hunk 3]
```diff
@@ -2,22 +2,63 @@ import type { EventHandlerDefinition, LambdaContext } from '../types.js'
...
+    messageGroupId: string
+    records: {
+        payload: Try<Payload>
+        raw: SQSRecord
+        item: number
+    }[]
+}
+
+export interface SQSMessageGrouping {
+    by?: 'messageGroupId'
```
Comment (pevisscher) [inline src/events/sqs/types.ts:22]:
```suggestion
    by?: 'message-group-id'
```

**Review by pevisscher — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (MickVanDuijn) [reply package.json:62]: we need to update this one in `therefore` first, I'm getting conflicting dependency resolution

Comment (MickVanDuijn) [reply package.json]: updated!

**Review by pevisscher — APPROVED** [review]

**Pull request merged**